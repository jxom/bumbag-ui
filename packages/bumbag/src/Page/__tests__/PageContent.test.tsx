import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { PageContent } from '../index';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<PageContent>Hello world</PageContent>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<PageContent color="primary">Hello world</PageContent>);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('variants', () => {
  it('styles.base should render correctly', () => {
    const { container } = render(<PageContent variant="test">hello world</PageContent>, {
      theme: {
        PageContent: {
          variants: { test: { styles: { base: { backgroundColor: 'red' } } } },
        },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('overrides', () => {
  it('styles.base should render correctly', () => {
    const { container } = render(
      <PageContent
        overrides={{
          PageContent: { styles: { base: { backgroundColor: 'red' } } },
        }}
      >
        hello world
      </PageContent>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('styles.base should render correctly', () => {
    const { container } = render(<PageContent>hello world</PageContent>, {
      theme: {
        PageContent: { styles: { base: { backgroundColor: 'red' } } },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<PageContent>hello world</PageContent>, {
      theme: {
        PageContent: { defaultProps: { className: 'test', color: 'primary' } },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
