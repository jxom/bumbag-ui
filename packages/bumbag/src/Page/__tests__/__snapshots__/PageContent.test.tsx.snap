// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  padding: 2rem 1rem;
}

@media screen and (max-width: 767px) {
  .emotion-3 {
    padding-top: 1rem;
    padding-bottom: 1rem;
  }
}

.emotion-4 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width: 768px) {
  .emotion-4 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

.emotion-4 {
  max-width: 768px;
}

.emotion-6 {
  color: var(--bb-palette-primary, #000000)!important;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-PageContentWrapper"
>
  <div
    class="emotion-3 emotion-4 emotion-1 emotion-6 test bb-Box bb-Container bb-PageContent"
  >
    hello world
  </div>
</div>
`;

exports[`overrides styles.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  padding: 2rem 1rem;
}

@media screen and (max-width: 767px) {
  .emotion-3 {
    padding-top: 1rem;
    padding-bottom: 1rem;
  }
}

.emotion-3 {
  background-color: var(--bb-palette-red, #ff0000);
}

.emotion-4 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width: 768px) {
  .emotion-4 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

.emotion-4 {
  max-width: 768px;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-PageContentWrapper"
>
  <div
    class="emotion-3 emotion-4 emotion-1 emotion-2 bb-Box bb-Container bb-PageContent"
  >
    hello world
  </div>
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  padding: 2rem 1rem;
}

@media screen and (max-width: 767px) {
  .emotion-3 {
    padding-top: 1rem;
    padding-bottom: 1rem;
  }
}

.emotion-4 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width: 768px) {
  .emotion-4 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

.emotion-4 {
  max-width: 768px;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-PageContentWrapper"
>
  <div
    class="emotion-3 emotion-4 emotion-1 emotion-2 bb-Box bb-Container bb-PageContent"
  >
    Hello world
  </div>
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  padding: 2rem 1rem;
}

@media screen and (max-width: 767px) {
  .emotion-3 {
    padding-top: 1rem;
    padding-bottom: 1rem;
  }
}

.emotion-4 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width: 768px) {
  .emotion-4 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

.emotion-4 {
  max-width: 768px;
}

.emotion-6 {
  color: var(--bb-palette-primary, #000000)!important;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-PageContentWrapper"
>
  <div
    class="emotion-3 emotion-4 emotion-1 emotion-6 bb-Box bb-Container bb-PageContent"
  >
    Hello world
  </div>
</div>
`;

exports[`theming styles.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  padding: 2rem 1rem;
}

@media screen and (max-width: 767px) {
  .emotion-3 {
    padding-top: 1rem;
    padding-bottom: 1rem;
  }
}

.emotion-3 {
  background-color: var(--bb-palette-red, #ff0000);
}

.emotion-4 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width: 768px) {
  .emotion-4 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

.emotion-4 {
  max-width: 768px;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-PageContentWrapper"
>
  <div
    class="emotion-3 emotion-4 emotion-1 emotion-2 bb-Box bb-Container bb-PageContent"
  >
    hello world
  </div>
</div>
`;

exports[`variants styles.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  padding: 2rem 1rem;
}

@media screen and (max-width: 767px) {
  .emotion-3 {
    padding-top: 1rem;
    padding-bottom: 1rem;
  }
}

.emotion-3 {
  background-color: var(--bb-palette-red, #ff0000);
}

.emotion-4 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width: 768px) {
  .emotion-4 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

.emotion-4 {
  max-width: 768px;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-PageContentWrapper"
>
  <div
    class="emotion-3 emotion-4 emotion-1 emotion-2 bb-Box bb-Container bb-PageContent"
  >
    hello world
  </div>
</div>
`;
